cmake/modules/extensions.cmake

```diff
@@ -2536,6 +2536,47 @@ function(zephyr_get variable)
   endif()
 endfunction(zephyr_get variable)
 
+# todo: add description
+function(zephyr_get_list variable)
+  cmake_parse_arguments(GET_VAR "" "SYSBUILD" "" ${ARGN})
+
+  if(DEFINED GET_VAR_SYSBUILD)
+    if(NOT (${GET_VAR_SYSBUILD} STREQUAL "GLOBAL" OR
+            ${GET_VAR_SYSBUILD} STREQUAL "LOCAL")
+    )
+      message(FATAL_ERROR "zephyr_get_list(... SYSBUILD) requires GLOBAL or LOCAL.")
+    endif()
+  else()
+    set(GET_VAR_SYSBUILD "GLOBAL")
+  endif()
+
+  if(SYSBUILD)
+    get_property(sysbuild_name TARGET sysbuild_cache PROPERTY SYSBUILD_NAME)
+    get_property(sysbuild_main_app TARGET sysbuild_cache PROPERTY SYSBUILD_MAIN_APP)
+    get_property(sysbuild_${variable} TARGET sysbuild_cache PROPERTY ${sysbuild_name}_${variable})
+    if(NOT DEFINED sysbuild_${variable} AND
+       (${GET_VAR_SYSBUILD} STREQUAL "GLOBAL" OR sysbuild_main_app)
+    )
+      get_property(sysbuild_${variable} TARGET sysbuild_cache PROPERTY ${variable})
+    endif()
+  endif()
+
+  get_property(from_sysbuild TARGET sysbuild_cache PROPERTY ${variable})
+    list(APPEND ${variable} ${${variable}})
+    list(APPEND ${variable} ${from_sysbuild})
+
+  if(DEFINED sysbuild_${variable})
+    list(APPEND ${variable} ${sysbuild_${variable}})
```

Comment:
Added some comments, and yes. I have actually flipped the order to make the sysbuild setting be last when the MERGE option is used, so that priority of the image-specific configuration is higher